### PR TITLE
Create aleung18.md

### PR DESCRIPTION
**Pull request by aleung18**
## Describe your project in 1-3 sentences

I am going to build a pet feeder for my dog using a Raspberry Pi, stepper motors, and driver boards. I am going to CAD design the feeder and print the components with my 3D printer. In the future, I would like to add the Raspberry Pi Camera module so that pet owners can see their dog eat when they are away. I am excited to build it because I will no longer have to feed my dog if successful! (Just kidding). Jokes aside, I am excited because it is my first “real” hardware project, and I have been trying out CAD design and playing with circuit boards and would like to make a project that has reflected what I have learned so far.

## Hack Club Slack handle
Ashton Leung

## Checklist before submitting proposal:

- [x] I am a current high school student (or younger).
- [x] I have submitted an RSVP at <https://hackclub.com/winter>
- [x] I am a member of the Hack Club Slack (Join at <https://hackclub.com/slack>)
- [x] I have self-reviewed my project proposal.

### CODE REVIEW

**Review by cfanoulis — CHANGES_REQUESTED** [review]
Hello! Great to see that your project is benefitting your pet.

Can you quickly explain how you plan on using motors to allow for automatic feeding?

**Comment by aleung18** [comment]
> Hello! Great to see that your project is benefitting your pet.
> 
> Can you quickly explain how you plan on using motors to allow for automatic feeding?

I am going to place the stepper motor underneath the drum of the 3D printed contraption, and connect it to a driver board, which will be connected to an external supply (positive and ground connected to breadboard, which is connected to 5-12V power supply), and darlington array will be connected to the Raspberry Pi. The stepper motor will rotate a section of the contraption (design is modular, different parts of model are stacked on top of each other) to dispense the treat for the pet.

**Comment by cfanoulis** [comment]
Great! And just a sanity check, you have included shipping fees and taxes right?

**Comment by aleung18** [comment]
> Great! And just a sanity check, you have included shipping fees and taxes right?

Yes, they're both included!

**Comment by cfanoulis** [comment]
Great, one last thing. Is this plan all laid out by yourself, or are you following a guide? If you are, please add a link in your RFC

**Comment by aleung18** [comment]
> Great, one last thing. Is this plan all laid out by yourself, or are you following a guide? If you are, please add a link in your RFC

I am going to follow a guide, but I want to venture out a little and try to create my own design for the pet feeding contraption. I have added the link to my RFC!

**Comment by cfanoulis** [comment]
Amazing, thanks for that
Some coffee later, I did notice you talk about 3D-printed parts. Do you have your own 3D printer, or will you order it from somewhere?

If you do have a 3D printer, would you maybe need extra filament?

**Comment by aleung18** [comment]
> Amazing, thanks for that Some coffee later, I did notice you talk about 3D-printed parts. Do you have your own 3D printer, or will you order it from somewhere?
> 
> If you do have a 3D printer, would you maybe need extra filament?
> 
> Also, I do notice that this is your first hardware project (congrats!), do you have a power supply for your Pi?

Yes, I have a 3D printer and I think I should have enough filament! I do have a power supply for my Pi as well!

**Comment by cfanoulis** [comment]
Perfect, care to add them to your budget table as well? (with a price of 0$)

**Comment by aleung18** [comment]
> Perfect, care to add them to your budget table as well? (with a price of 0$)

Will do!

File: aleung18.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+## Budget
+
+What materials will you need for your project? Where will you get them? How much does it cost? Please include all materials, including components you already own. Make sure to factor in shipping costs and sales tax.
+
+| Product                            | Supplier/Link                                           | Cost   |
+| ---------------                    | --------------------------------------------------------| ------ |
+| Ender 3 V2 (already own)           |                                                         | $0     |
+| Overture Filament (already own)    |                                                         | $0     |
+| Breadboard & Battery Power Supply (already own)                                              | $0     |
```
Comment (cfanoulis) [inline aleung18.md:39]: minor nit - just to make sure the middle column stays empty :)

```suggestion
| Breadboard & Battery Power Supply (already own)                                              || $0     |
```

**Review by cfanoulis — CHANGES_REQUESTED** [review]
one laaaaaast thing i swear - its a formatting thing

**Review by cfanoulis — APPROVED** [review]
And you're set to make! A banker from @hackclub/bank-ops will be in touch shortly to get you setup with your grant!

And go get your doggo a treat (maybe send us a photo in #surroundings?)

**Comment by aleung18** [comment]
> And you're set to make! A banker from @hackclub/bank-ops will be in touch shortly to get you setup with your grant!
> 
> And go get your doggo a treat (maybe send us a photo in #surroundings?)

Thank you Charalampos! I'll stop by the pet store after school!

**Review by hackliv — APPROVED** [review]
Hey @aleung18 congrats on posting your PR for your first ever hardware project!! I'm so excited to see you make this and hopefully your dog likes it too. 😄 I'm dming you on Slack now to set up some time to meet and chat about the grant more!

**Pull request merged**